### ISSUE TITLE
NVIDIA Invalid Result GPU ID 0 - with latest version

### ISSUE THREAD

**Issue by Arramis** [fireice-uk/xmr-stak#1594]
Please provide as much as possible information to reproduce the issue.

# Basic information
  - Type of the CPU. - AMD RYZEN 1700x
  - Type of the GPU (if you try to miner with the GPU). EVGA GFORCE 1080 FTW ACX 3.0 08G-P4-6286-KR

# Issue with the execution
  - Do you compiled the miner by our own? 
NO - downloaded from here.

  run `./xmr-stak --version-long` and add the output here
 xmr-stak/2.4.3/26a5d65f/master/win/nvidia-amd-cpu/aeon-cryptonight-monero/20

# Stability issue
  - Is the CPU or GPU overclocked?
NO
  - Is the Main memory of the CPU or GPU undervolted?
I do not know - i did not modify anything.

#error 
RESULT REPORT
Difficulty       : 4000
Good results     : 33 / 35 (94.3 %)
Avg result time  : 2.5 sec
Pool-side hashes : 92000

Top 10 best results found:
|  0 |            96082 |  1 |            43653 |
|  2 |            27547 |  3 |            24195 |
|  4 |            23085 |  5 |            21177 |
|  6 |            18143 |  7 |            16451 |
|  8 |            15700 |  9 |            12295 |

Error details:
| Count | Error text                       | Last seen           |
|     2 | NVIDIA Invalid Result GPU ID 0   | 2018-05-24 23:09:12 |

This error appeared after windows update. Never had any issue with xmr-stak-win64 or xmr-stak -nvidia (the latest version before win64).
I played a bit with b-factor and other parameters and the best succes rate is arround 95%.

This is nvidia.txt

"gpu_threads_conf" :
[
  // gpu: GeForce GTX 1080 architecture: 61
  //      memory: 6758/8192 MiB
  //      smx: 20
  { "index" : 0,
    "threads" : 32, "blocks" : 50,
    "bfactor" : 10, "bsleep" :  100,
    "affine_to_cpu" : true, "sync_mode" : 0,
  },

],

i tried without affine_to_cpu
i tried with lower and higher threads, blocks
i tried bfactor from 1 to 15 and sleep from 25 to 100
i tried syncmode 0 and 3.

Any ideas? :)
Thank you.

**Comment by Loule-Corro** [comment]
Same here, maybe last cumulative update for Win 10 1803 ?

**Comment by Arramis** [comment]
yep, that update :) (1803 build 17134.48), so i`m not the only one.

**Comment by JerichoJones** [comment]
Have you tried uninstall/reinstall driver after the update?

**Comment by Arramis** [comment]
Yes, no solution.

**Comment by jirkag** [comment]
I have same problem on linux UBUNTU 18.04 with AMD 18.20 drivers. I am getting invalid results for all my AMD cards in rig. Problem disaper in case that I am revert back to 17. 50 drivers. Please look to this issue, never cards are working with new drivers only.

**Comment by smash120** [comment]
I had the same issue with my AMD cards and after playing around with my miner and pc for a few days I deleted all files in the following directory C:\Users\userprofile\AppData\Local\.openclcache and ran the miner again and the AMD Invalid Result GPU ID 0 no longer displayed.

**Comment by Arramis** [comment]
Thanks but i do not have the directory Local.openclcache. :)

**Comment by serggray** [comment]
Please install old Nvidia drivers v380 and problem will vanished.

**Comment by Spudz76** [comment]
Current Release is built for CUDA10.0
Current Drivers have CUDA 10.1
Run a Driver that still contains CUDA10.0 something must be broken in the backward-compatibility (like every time)
Or, compile against CUDA Toolkit 10.1 and run latest driver

You want the detection line `[x/y]` x and y to be equal
It will run anyway as long as CUDA still allows it but it is never ideal.
